src/commands/redis_cmd.cc

```diff
@@ -5304,6 +5308,9 @@ class CommandClusterX : public Commander {
     } else {
       *output = Redis::Error("Invalid cluster command options");
     }
+    if (needPersistNodesInfo) {
```

Comment:
@git-hulk What do you think about hiding this "knowledge" inside `Cluster`? I mean that the `Cluster` class calls the `DumpClusterNodes` function inside ` SetNodeId`/`SetSlot`/`SetClusterNodes` and commands know nothing about `NodesFilePath` and all the internals about storing the cluster topology in a file, etc.

Reply:
Yes, it makes sense. Will resolve it later.

Reply:
Seems that command will failed if `DumpClusterNodes` failed. However, the cluster info is already updated.

Reply:
re @mapleFU

Yes, cannot promise the atomic now.

Reply:
re @torwig, I recalled why to specify the path here. It will convenient for adding test cases without depending on server->config, as well as avoid modifying a lot of previous test cases.